xpad.c

```diff
@@ -1971,6 +2064,26 @@ static int xpad_probe(struct usb_interface *intf, const struct usb_device_id *id
 
 	usb_set_intfdata(intf, xpad);
 
+	/* Special packet type detection */
+	if (le16_to_cpu(udev->descriptor.idVendor) == 0x045e) { /* Microsoft controllers */
+		if (le16_to_cpu(udev->descriptor.idProduct) == 0x02e3) { /* Elite 1 controller */
+			/* The original elite controller always uses the oldest type of extended packet */
+			xpad->ptype = PTYPE_EONE;
+		} else if (le16_to_cpu(udev->descriptor.idProduct) == 0x0b00) { /* Elite 2 controller */
+			/* The elite 2 controller has seen multiple packet revisions. These are tied to specific firmware versions */
+			if (le16_to_cpu(udev->descriptor.bcdDevice) < 0x0500) {
+				/* This is the format that the Elite 2 used prior to the BLE update */
+				xpad->ptype = PTYPE_ETWOOLD;
+			} else if (le16_to_cpu(udev->descriptor.bcdDevice) < 0x050b) {
+				/* This is the format that the Elite 2 used prior to the update that split the packet */
+				xpad->ptype = PTYPE_ETWO5;
```

Comment:
This is another configuration that is impossible for me to test due to lack of access to the firmware in question. I'm not even 100% sure if this packet format actually existed... I'm mostly just taking SDL's word for it... but, if it _did_ exist, it definitely stopped existing when the BLE update happened (v5.11).

Once again, I think this is a situation where we'll just have to wait and see if anyone with a weird firmware version happens to come out of the woodwork and report an issue.